pipelines/rj_smas__disparo_pic/utils/api_handler.py

```
@@ -0,0 +1,130 @@
+
+            if token:
+                self.token = token
+                self.headers["Authorization"] = f"{self.token_type} {token}"
+                log("Authentication token obtained successfully")
+            else:
+                log("Warning: No token found in login response")
+                log(f"Response keys: {list(response_data.keys())}")
+
+        except requests.RequestException as e:
+            log(f"Login failed: {e}")
+            raise Exception(f"Failed to authenticate with API: {e}")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)